src/backends/mock_main.rs

```diff
@@ -265,6 +267,14 @@ impl MockMainPod {
     }
 }
 
+pub fn hash_statements(statements: &[middleware::Statement]) -> Result<middleware::Hash> {
+    let field_elems = statements
+        .into_iter()
+        .flat_map(|statement| statement.clone().to_fields().0)
+        .collect::<Vec<_>>();
+    Ok(Hash(PoseidonHash::hash_no_pad(&field_elems).elements))
```

Comment:
Should we use a padding here?  Maybe we can revisit this in the future (for all hashes, not just this one)